### DISCUSSION TITLE
Which data is used for learning bayes or fuzzy storage?

### DISCUSSION THREAD

**Discussion by syntaxys** [rspamd/rspamd#5231] [General]
Hi,

actually I reject messages above score 12 but doing also an export of the mail to postmasters junk folder as backup purpose for some days.
In this case, these messages are loosing their original „Envelope From“,  „Envelope To“ and the „Received“ Headers. Instead, these headers are available, e.g.:
```
Return-Path: <postmaster@mydomain.tld>
Delivered-To: mbox_admin@mydomain.tld
Received: from rspamd.mydomain.tld ([192.168.123.2])
        by mailstore.mydomain.tld with LMTP
        id mDmYAC0QP2fkOAkADDlYHw
        (envelope-from <postmaster@mydomain.tld>)
        for <mbox_admin@mydomain.tld>; Thu, 21 Nov 2024 11:49:17 +0100
Received: from rspamd.mydomain.tld (localhost [127.0.0.1])
        by mta.mydomain.tld (Postfix) with SMTP id F16417D0CF
        for <quarantine@mydomain.tld>; Thu, 21 Nov 2024 11:49:16 +0100 (CET)
```
So my question is, if I use such mail files for manual learning, will these headers also be used for hashing? Is there anywhere an information about which objects that will be used for hashing? 

Thanks for your help, kind regards
Achim

**Comment by fatalbanana** [comment]
Bayes uses textual content of the message (Subject & message body).

You can use `rspamadm mime ex -w foo.eml` to view this content.

**Reply by fatalbanana** [reply]
Fuzzy operates on full message parts; fragments of text parts & embedded JS contained in PDF documents.

**Reply by syntaxys** [reply]
Thanks for your reply!

> Fuzzy operates on full message parts

Does this mean that headers relating to internal transport are also taken into account for hashing? Can I prevent this?

**Reply by fatalbanana** [reply]
No, headers are not used at all.

**Reply by moisseev** [reply]
> Bayes uses textual content of the message (Subject & message body).

Regarding Bayes:  

As seen in the code, there are [hardcoded defaults](https://github.com/rspamd/rspamd/blob/41eab5b874721f7abc23144e5c3386392f1820f7/lualib/lua_stat.lua#L531C1-L546C4 ):  

```lua  
  -- Default  
  local res_config = {  
    classify_headers = {  
      "User-Agent",  
      "X-Mailer",  
      "Content-Type",  
      "X-MimeOLE",  
      "Organization",  
      "Organisation"  
    },  
    classify_images = true,  
    classify_mime_info = true,  
    classify_urls = true,  
    classify_meta = true,  
    classify_max_tlds = 10,  
  }  
``` 

And the [configuration default](https://github.com/rspamd/rspamd/blob/41eab5b874721f7abc23144e5c3386392f1820f7/conf/options.inc#L40C1-L45C3):  

```ucl  
classify_headers = [  
	"User-Agent",  
	"X-Mailer",  
	"Content-Type",  
	"X-MimeOLE",  
];  
``` 

So, by default, besides the textual content, the following headers are considered:  
- `User-Agent`  
- `X-Mailer`  
- `Content-Type`  
- `X-MimeOLE`  

Additionally, Bayes processes metadata like URLs, MIME info, and image-related data.
